x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/logical/Not.java

```diff
@@ -58,8 +57,8 @@ public String processScript(String script) {
     @Override
     protected Expression canonicalize() {
         Expression canonicalChild = field().canonical();
-        if (canonicalChild instanceof Negateable) {
-            return ((Negateable) canonicalChild).negate();
+        if (canonicalChild instanceof UnaryNegateable) {
```

Comment:
This doesn't look good as it essentially removes `NOT` over binary logical negateable `NOT (x AND y)`.

Reply:
That's a mistake for sure, thx for catching!

Reply:
Can you add some tests please with `NOT` in `HAVING` over AND and IS NULL to check this automatically in the future?